tests/test_matching_algorithms/test_pay_as_bid_matching_algorithm.py

```diff
@@ -46,15 +56,129 @@ def test_perform_pay_as_bid_match(self):
+             "offers": [{"id": 11, "seller": "B", "energy_rate": 1, "energy": 60}],
+             "selected_energy": 10, "trade_rate": 1.5},
+
+        ]
+        assert trades == expected_trades
+
+    @staticmethod
+    def test_perform_pay_as_bid_match_single_offer_bid():
+        """
+        Test whether a single offer can match with a bid.
+        """
+        data = {
+            "market1": {
+                "2021-10-06T12:00": {
+                    "bids": [{"id": 1, "buyer": "B", "energy_rate": 3, "energy": 20}],
+                    "offers": [{"id": 2, "seller": "A", "energy_rate": 3.1, "energy": 30}],
+                }
+            },
+        }
+        trades = PayAsBidMatchingAlgorithm.get_matches_recommendations(data)
+        assert not trades
+        # Adapting the offer energy rate to a value that can match with the bid
+        data["market1"]["2021-10-06T12:00"]["offers"][0]["energy_rate"] = 1
+        trades = PayAsBidMatchingAlgorithm.get_matches_recommendations(data)
+        expected_trades = [
+            {"market_id": "market1",
+             "time_slot": "2021-10-06T12:00",
+             "bids": [{"id": 1, "buyer": "B", "energy_rate": 3, "energy": 20}],
+             "offers": [{"id": 2, "seller": "A", "energy_rate": 1, "energy": 30}],
+             "selected_energy": 20, "trade_rate": 3}
+        ]
+        assert trades == expected_trades
+
+    @staticmethod
+    def test_perform_pay_as_bid_match_single_offer_multiple_bids():
+        """
+        Test whether a single offer can match with multiple bids.
+        """
+        data = {
+            "market1": {
+                "2021-10-06T12:00": {
+                    "bids": [
+                        {"id": 1, "buyer": "B", "energy_rate": 3, "energy": 20},
+                        {"id": 2, "buyer": "C", "energy_rate": 3, "energy": 20},
+                        {"id": 3, "buyer": "A", "energy_rate": 3, "energy": 20},
+                    ],
+                    "offers": [{"id": 4, "seller": "A", "energy_rate": 1, "energy": 70}],
+                }
+            },
+        }
+        trades = PayAsBidMatchingAlgorithm.get_matches_recommendations(data)
+        # The bid-id 3 will not match because it has the same buyer as the seller
+        expected_trades = [
+            {"market_id": "market1",
+             "time_slot": "2021-10-06T12:00",
+             "bids": [{"id": 2, "buyer": "C", "energy_rate": 3, "energy": 20}],
+             "offers": [{"id": 4, "seller": "A", "energy_rate": 1, "energy": 70}],
+             "selected_energy": 20, "trade_rate": 3},
+            {"market_id": "market1",
+             "time_slot": "2021-10-06T12:00",
+             "bids": [{"id": 1, "buyer": "B", "energy_rate": 3, "energy": 20}],
+             "offers": [{"id": 4, "seller": "A", "energy_rate": 1, "energy": 70}],
+             "selected_energy": 20, "trade_rate": 3},
+        ]
+        assert trades == expected_trades
```

Comment:
I was surprised to see trades here, so I propose to rename in the whole module:
```suggestion
        recommendations = PayAsBidMatchingAlgorithm.get_matches_recommendations(data)
        # The bid-id 3 will not match because it has the same buyer as the seller
        expected_recommendations = [
            {"market_id": "market1",
             "time_slot": "2021-10-06T12:00",
             "bids": [{"id": 2, "buyer": "C", "energy_rate": 3, "energy": 20}],
             "offers": [{"id": 4, "seller": "A", "energy_rate": 1, "energy": 70}],
             "selected_energy": 20, "trade_rate": 3},
            {"market_id": "market1",
             "time_slot": "2021-10-06T12:00",
             "bids": [{"id": 1, "buyer": "B", "energy_rate": 3, "energy": 20}],
             "offers": [{"id": 4, "seller": "A", "energy_rate": 1, "energy": 70}],
             "selected_energy": 20, "trade_rate": 3},
        ]
        assert recommendations == expected_recommendations
```
Thanks!